tests/end_to_end/test_create_new_template.py

```diff
@@ -0,0 +1,188 @@
+import datetime
+import os
+import re
+import uuid
+
+from playwright.sync_api import expect
+
+E2E_TEST_URI = os.getenv("NOTIFY_E2E_TEST_URI")
+
+
+def _setup(page, end_to_end_context):
+    # Prepare for adding a new service later in the test.
+    current_date_time = datetime.datetime.now()
+    new_service_name = "E2E Federal Test Service {now} - {browser_type}".format(
+        now=current_date_time.strftime("%m/%d/%Y %H:%M:%S"),
+        browser_type=end_to_end_context.browser.browser_type.name,
+    )
```

Comment:
It looks like we might be able to leverage just the `page` object passed in and not need the `end_to_end_context` fixture here, which is just duplicative anyway and if anything could cause weird syncing or session issues because you're now working with two separate `BrowserContext` objects.

You can get the existing `BrowserContext` off of the `page` object by just referencing [`page.context`](https://playwright.dev/python/docs/api/class-page#page-context) it looks like, which should be able to provide you the same info ultimately:  `page.context.browser.browser_type.name` most likely, and that'll make sure you're using the existing context that you're starting the tests with. 🙂 

Remember too that in this case, `page` (since it's being passed in from the test method itself below) is ultimately referring to an authenticated browser context as well, which is handling the logging in of the test user!